### DISCUSSION TITLE
Import sort: automatically detect third party packages with same base package as first party

### DISCUSSION THREAD

**Discussion by ivan94fi** [astral-sh/ruff#9226] [Q&A]
Hi, I have multiple python packages with the same first name, for example `company.package1`, `company.package2`, etc.
When in `company.package1` Isort automatically flags `company.package2` import as a third party, while Ruff detects every package starting with `company` as first party, changing the sort order.

If I use the following configuration:
```toml
[tool.ruff]
src = ["src"]

[tool.ruff.lint.isort]
known-third-party = ["company.package2"]
```
then `company.package2` is correctly formatted as a third party package, but this is not scalable as I have many of these packages and I would need to configure every package differently and update every one with every new package created.

I also tried setting `known-third-party = ["company.*"]`, but this makes all company imports as third party, even the first party ones.

Is there some option to configure Ruff the way I need? Thank you.

**Comment by hamedrabah** [comment]
This is the same-package heuristic rather than `src` discovery. By default, when Ruff analyzes a file inside `company.package1`, any import whose first package segment is `company` can be classified as first-party.

Disable that heuristic and let the full-path `src` lookup decide:

```toml
[tool.ruff]
src = ["src"]

[tool.ruff.lint.isort]
detect-same-package = false
```

For `import company.package2`, Ruff's source-root check requires the full `company/package2` path to exist under this project's `src`. If only `company/package1` exists, package2 then falls through to third-party, while package1 remains first-party.

This avoids maintaining `known-third-party` for every sibling namespace package. If multiple sibling packages are present in the same checkout, however, they really are all visible first-party sources to Ruff; in that case explicit sections or separate project configurations are necessary.
